functions.inc.php

```diff
@@ -1076,11 +1077,14 @@ function _pacrypt_dovecot($pw, $pw_db = '')
         throw new Exception("invalid dovecot encryption method");
     }
 
-    # digest-md5 hashes include the username - until someone implements it, let's declare it as unsupported
+    $doveadm_options = '';
+
     if (strtolower($method) == 'digest-md5') {
-        throw new Exception("Sorry, \$CONF['encrypt'] = 'dovecot:digest-md5' is not supported by PostfixAdmin.");
+        if (empty($username)) {
+            throw new Exception("\$CONF['encrypt'] = 'dovecot:digest-md5' require username.");
+        }
+        $doveadm_options = ' -u ' . $username;
```

Comment:
$username needs to pass through escapeshellarg?

Reply:
fix